scss/_layouts_application.scss

```diff
@@ -0,0 +1,170 @@
+    grid-template-rows: min-content 1fr min-content;
+    height: 100vh;
+    width: 100vw;
+  }
+
+  $panel-drop-shadow: 0 0 2rem 0 rgba($color-x-dark, $shadow-opacity);
+  $panel-drop-shadow-transparent: 0 0 0 0 transparent;
+
+  // Top navigation bar to toggle side navigation on small screens
+  .l-navigation-bar {
+    grid-area: navbar;
+  }
+
+  // Navigation panel/drawer
+  .l-navigation {
+    bottom: 0;
+    box-shadow: $panel-drop-shadow;
+    height: 100vh;
+    left: 0;
+    overflow-y: auto;
+    position: fixed;
+    top: 0;
+    width: 100%;
+    z-index: 103; // consistent with side-navigation component, above global nav
+
+    @media (min-width: $breakpoint-x-small) {
+      width: auto;
+    }
+
+    &.is-collapsed {
+      width: 0;
+    }
+  }
+
+  $layout-navigation-collapsed-width: 3rem;
+  $layout-navigation-expanded-width: 15rem;
+
+  .l-navigation__drawer {
+    height: 100vh;
+    width: auto;
+
+    @media (min-width: $breakpoint-x-small) {
+      width: $layout-navigation-expanded-width;
+    }
+  }
+
+  @media (min-width: $breakpoint-medium) {
+    .l-navigation-bar {
+      // use navigation bar as a grid spacer for collapsed navigation
+      grid-area: nav;
+      overflow: hidden;
+      width: $layout-navigation-collapsed-width;
+    }
+
+    .l-navigation {
+      box-shadow: $panel-drop-shadow-transparent;
+      transition: width 100ms, box-shadow 100ms;
+      width: $layout-navigation-collapsed-width;
+
+      &.is-collapsed {
+        width: $layout-navigation-collapsed-width;
+      }
+    }
+
+    .l-navigation:hover {
+      box-shadow: $panel-drop-shadow;
+      width: $layout-navigation-expanded-width;
+    }
+
+    .l-navigation.is-pinned {
+      box-shadow: $panel-drop-shadow-transparent;
+      grid-area: nav;
+      position: static;
+      width: $layout-navigation-expanded-width;
+    }
+  }
+
+  @media (min-width: $breakpoint-large) {
+    .l-navigation-bar {
+      // hide navigation bar completely when not needed
+      display: none;
+    }
+
+    .l-navigation {
+      box-shadow: none;
+      grid-area: nav;
+      position: static;
+      width: $layout-navigation-expanded-width;
+
+      &:hover {
+        box-shadow: none;
+      }
+
+      &.is-collapsed {
+        width: $layout-navigation-expanded-width;
+      }
+    }
+  }
+
+  .l-main {
+    grid-area: main;
+    overflow-y: auto;
+  }
+
+  .l-status {
+    border-top: 1px solid $colors--light-theme--border-low-contrast;
+    grid-area: status;
+    z-index: 102;
+  }
+
+  .l-aside {
+    box-shadow: $panel-drop-shadow;
+    grid-area: main;
+    justify-self: right;
+    overflow-y: auto;
+    z-index: 101; // below side nav, above anything else
+
+    &.is-pinned {
+      border-left: 1px solid $colors--light-theme--border-low-contrast;
+      box-shadow: none;
+      grid-area: aside;
+      justify-self: auto;
+    }
+  }
+
+  @include vf-l-application-panels;
+
+  // minimal demo styling
+  .l-application.app-demo {
```

Comment:
Can't these be dropped in the example mark up?

Reply:
Or are they required for the layout to work in practice?

Reply:
@anthonydillon  I don't like them here either, cause they are solely for the purpose of the structure example.
I kept it here similar to [`.grid-demo`](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/scss/_patterns_grid.scss#L48-L51) (although I'm not so keen on grid 'demo' styles in Vanilla either), and to make it easier to reuse SCSS variables and breakpoints.

In the meantime I managed to get rid of breakpoints and use `u-hide` classes in examples instead, so technically it could go into example HTML (assuming we hardcode colors and paddings in there).

If you rather have it in demo, I'm fine with moving it out to HTML (especially that they are for one example only, not shared anywhere).

Reply:
Can you just hardcode these styling in the example HTML header and not have it appear in the example code so people don't copy them? We need to be careful with styling in examples that are not from the framework as it could lead to unexpected results if just copying the mark up.

Reply:
If we have the grid-demo in the framework, lets leave it and think of a better solution separately

Reply:
Maybe a separate "demo" stylesheet, build from SCSS, but not part of Vanilla framework `build.scss`?
We will need to make sure it's available to examples and codepens, but it will have benefits of having access to variables in SCSS, while being explicitly separate from the framework?

I'll create an issue for that.